src/lib/sparse_ledger_lib/sparse_ledger.ml

```diff
@@ -181,13 +186,16 @@ end = struct
       | Node _, [] ->
           failwith "path is shorter than a tree's branch"
     in
-    traverse_through_nodes (tree0, List.rev path0)
+    let (`Leaf_to_root path_to_root) = path0 in
+    let path_to_leaf = List.rev path_to_root in
```

Comment:
I like this. It was so confusing to me until I found this `List.rev` here. It's better to give it a proper name.